cmd/podman/create.go

```diff
@@ -168,21 +170,43 @@ func createContainer(c *cliconfig.PodmanCommand, runtime *libpod.Runtime) (*libp
 			imageName = newImage.ID()
 		}
 
-		// add healthcheck if it exists AND is correct mediatype
-		_, mediaType, err := newImage.Manifest(ctx)
-		if err != nil {
-			return nil, nil, errors.Wrapf(err, "unable to determine mediatype of image %s", newImage.ID())
+		var healthCheckCommandInput string
+		// if the user disabled the healthcheck with "none", we skip adding it
+		if c.Flags().Lookup("healthcheck-command") != nil {
+			healthCheckCommandInput = c.String("healthcheck-command")
 		}
-		if mediaType == manifest.DockerV2Schema2MediaType {
-			healthCheck, err = newImage.GetHealthCheck(ctx)
-			if err != nil {
-				return nil, nil, errors.Wrapf(err, "unable to get healthcheck for %s", c.InputArgs[0])
-			}
-			if healthCheck != nil {
-				hasHealthCheck = true
+
+		// the user didnt disable the healthcheck but did pass in a healthcheck command
+		// now we need to make a healthcheck from the commandline input
+		if healthCheckCommandInput != "none" && len(healthCheckCommandInput) > 0 {
+			if len(healthCheckCommandInput) > 0 {
+				healthCheck, err = makeHealthCheckFromCli(c)
+				if err != nil {
+					return nil, nil, errors.Wrapf(err, "unable to create healthcheck")
+				}
+				if healthCheck != nil {
+					hasHealthCheck = true
+				}
+			} else if len(healthCheckCommandInput) == 0 {
```

Comment:
How will this ever trigger? It's inside an `if` block that verifies `healthCheckCOmmandInput` is over 0